src/client/MapleCharacter.java

```diff
@@ -1806,14 +1795,28 @@ public void saveToDB(boolean dc, boolean fromcs) {
                     }
                 }
             }
-            changed_wishlist = false;
+			
+            if (changed_favorites) {
+                ps = con.prepareStatement("DELETE FROM favorites WHERE accountid = ?");
+                ps.setInt(1, getAccountID());
+                ps.executeUpdate();
+                ps.close();
+                for (int i : favorites) {
+                    ps = con.prepareStatement("INSERT INTO favorites(accountid, sn) VALUES(?, ?) ");
```

Comment:
It does make sense; prepared statements are made for efficiency. If there's more spots that are using this kind of building queries, they all should be refactored.
These things could reduce the speed a lot.